test/remote/gateways/remote_ixopay_test.rb

```diff
@@ -43,23 +43,23 @@ def test_failed_purchase
     assert_equal '2003', response.error_code
   end
 
-  def test_successful_authorize_and_capture
-    omit 'Not yet implemented'
-
+  def test_successful_authorize#_and_capture
     auth = @gateway.authorize(@amount, @credit_card, @options)
     assert_success auth
+    assert_equal 'FINISHED', auth.message
+    assert_not_nil auth.params['purchase_id']
+    assert_not_nil auth.params['reference_id']
 
-    assert capture = @gateway.capture(@amount, auth.authorization)
-    assert_success capture
-    assert_equal 'REPLACE WITH SUCCESS MESSAGE', capture.message
+    #assert capture = @gateway.capture(@amount, auth.authorization)
+    #assert_success capture
+    #assert_equal 'REPLACE WITH SUCCESS MESSAGE', capture.message
   end
 
   def test_failed_authorize
-    omit 'Not yet implemented'
-
     response = @gateway.authorize(@amount, @declined_card, @options)
     assert_failure response
-    assert_equal 'REPLACE WITH FAILED AUTHORIZE MESSAGE', response.message
+    assert_equal 'The transaction was declined', response.message
+    assert_equal 'ERROR',                        response.params['return_type']
```

Comment:
There might not be, but I'm wondering whether if there's an `error_code` returned that we could check here instead.

Reply:
we can check it also!